lib/rules/no-sync.js

```diff
@@ -27,25 +150,126 @@ module.exports = {
             },
         ],
         messages: {
-            noSync: "Unexpected sync method: '{{propertyName}}'.",
+            noSyncMember: "Unexpected sync method: '{{importName}}'.",
+            noSyncAlias:
+                "Unexpected sync method: '{{aliasName}}' ({{importName}}).",
         },
     },
 
     create(context) {
-        const selector =
-            context.options[0] && context.options[0].allowAtRootLevel
-                ? ":function MemberExpression[property.name=/.*Sync$/]"
-                : "MemberExpression[property.name=/.*Sync$/]"
+        const [options] = context.options
+        const syncRegex = /Sync$/u
 
         return {
-            [selector](node) {
-                context.report({
-                    node,
-                    messageId: "noSync",
-                    data: {
-                        propertyName: node.property.name,
-                    },
-                })
+            [ModuleDeclarations](node) {
+                const variables =
+                    context.sourceCode?.getDeclaredVariables?.(node) ??
+                    context.getDeclaredVariables?.(node)
+
+                for (const variable of variables) {
+                    for (const reference of variable.references) {
+                        if (
+                            options?.allowAtRootLevel === true &&
+                            isFunctionScope(reference.from) === false
+                        ) {
+                            continue
+                        }
+
+                        /** @type {import('eslint').Rule.Node} */
+                        const memberExpression = reference.identifier.parent
+                        if (
```

Comment:
nits: the name `memberExpression` seems a little confused to me - it's also likely to be a non-memberexpression node?

Reply:
Mm, I know what you mean.
I didn't really know what to call this, I called it memberExpression because I check to see if it's a memberExpression immediately after I create it 👀

Do you have any better name suggestions 😅

Reply:
something like `parentNode` or `identifierParentNode`?

Reply:
ok, given it's checked I think it's fine to be as-is. (naming is not an easy thing :). 😄